clickhouse/base/endpoints_iterator.cpp

```diff
@@ -0,0 +1,43 @@
+#include "endpoints_iterator.h"
+#include "../client.h"
+
+namespace clickhouse {
+
+RoundRobinEndpointsIterator::RoundRobinEndpointsIterator(const ClientOptions& opts) :
+     hosts (opts.hosts) 
```

Comment:
IMO you should put value from 'opts.host' here as first value, otherwise it is not used at all.